File: frontend/src/types/species.ts
export interface Order {
    id: string;
    name: string;
}

export interface Family {
    id: string;
    family_code: string;
    common_name: string;
    scientific_name: string;
    order: Order;
}

export interface Species {
    id: string;
    ebird_code: string;
    common_name: string;
    scientific_name: string;
    category: string;
    wildlife_group: string;

    // iNaturalist
    inat_taxon_id: number | null;
    image_url: string | null;
    thumbnail_url: string | null;
    image_license: string | null;
    image_attribution: string | null;
    image_source: string | null;

    family: Family;
}

export interface SpeciesExplorerSpecies {
    id: string;

    common_name: string;
    scientific_name: string;

    image_url: string | null;
    thumbnail_url: string | null;

    family_common_name: string | null;
    order_common_name: string | null;
}

export interface SpeciesExplorerResponse {
    items: SpeciesExplorerSpecies[];
    total: number;
    page: number;
    page_size: number;
}

export interface SpeciesListResponse {
    items: Species[];
    total: number;
    page: number;
    page_size: number;
}

File: backend/app/schemas/species/species.py
from __future__ import annotations

from uuid import UUID

from pydantic import BaseModel, ConfigDict


class OrderResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    name: str


class FamilyResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    family_code: str
    common_name: str
    scientific_name: str
    order: OrderResponse


class SpeciesResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    ebird_code: str
    common_name: str
    scientific_name: str
    category: str
    wildlife_group: str

    # iNaturalist
    inat_taxon_id: int | None
    image_url: str | None
    thumbnail_url: str | None
    image_license: str | None
    image_attribution: str | None
    image_source: str | None

    family: FamilyResponse


class SpeciesListResponse(BaseModel):
    items: list[SpeciesResponse]
    total: int
    page: int
    page_size: int